From 0a5616e7679bb6e917330d8aba8616fc8e44fbad Mon Sep 17 00:00:00 2001
From: Bartosz Bilas <b.bilas@grinn-global.com>
Date: Thu, 1 Jan 2025 00:00:00 +0000
Subject: [PATCH] uboot: Add Yocto config fragment merge support

Add support for merging the configuration fragment provided
by Yocto during the U-Boot build process.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>
---
 module/uboot/build.sh | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/module/uboot/build.sh b/module/uboot/build.sh
index f5c50ad..c94820c 100755
--- a/module/uboot/build.sh
+++ b/module/uboot/build.sh
@@ -25,7 +25,13 @@ build_uboot() {
     echo "CONFIG_BOOTCOMMAND=\"bootemmc android\"" >> $tmpfile
   fi
 
-  ./scripts/kconfig/merge_config.sh -m configs/${CONFIG_UBOOT_DEFCONFIG} $tmpfile
+  # Merging the configuration fragment provided by Yocto
+  if [ -f configs/yocto_fragment.cfg ]; then
+    ./scripts/kconfig/merge_config.sh -m configs/${CONFIG_UBOOT_DEFCONFIG} $tmpfile configs/yocto_fragment.cfg
+  else
+    ./scripts/kconfig/merge_config.sh -m configs/${CONFIG_UBOOT_DEFCONFIG} $tmpfile
+  fi
+
   mv .config configs/${TARGET_CONFIG}
   rm -fv $tmpfile
 
@@ -42,7 +48,7 @@ build_uboot() {
 
   eval "CONFIG_RDK_SYS=${CONFIG_RDK_SYS} $cmd2run make mrproper"
   if [ "is${CONFIG_RDK_SYS}" = "isy" ]; then
-    eval "CONFIG_RDK_SYS=${CONFIG_RDK_SYS} $cmd2run make ${CONFIG_UBOOT_DEFCONFIG} -j${CONFIG_CPU_NUMBER} V=1 HOSTCC=gcc CC=\"${CC}\""
+    eval "CONFIG_RDK_SYS=${CONFIG_RDK_SYS} $cmd2run make ${TARGET_CONFIG} -j${CONFIG_CPU_NUMBER} V=1 HOSTCC=gcc CC=\"${CC}\""
     eval "CONFIG_RDK_SYS=${CONFIG_RDK_SYS} $cmd2run make EXT_DTB=arch/${CONFIG_UBOOT_ARCH}/dts/${CONFIG_UBOOT_DTS}.dtb -j24 HOSTCC=gcc CC=\"${CC}\""
   else
     eval "CONFIG_RDK_SYS=${CONFIG_RDK_SYS} $cmd2run make ${TARGET_CONFIG} -j${CONFIG_CPU_NUMBER}"
-- 
2.30.2

